### PR TITLE
RFD 9: Advanced Session Control / Session Locking 

### PR DESCRIPTION
**Pull request by benarent**
Ported over content from a Google Spec -> RFD to better share advanced session control and session locking with a customer.

### CODE REVIEW

File: rfd/0009-advanced-session-control.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+This RFD provides a user locking mechanism to stop a user from continuing to access
+a system. This RFD will focus on how to disconnect and disable a user:
+
++ Teleport should disconnect and terminate any existing active sessions for the user.
++ Teleport should not allow creation of new sessions.
+
+## Why
+
+We've several request for advanced session control. These requests can be grouped
```
Comment (webvictim) [inline rfd/0009-advanced-session-control.md:23]:
```suggestion
We've had several requests for advanced session control. These requests can be grouped
```

File: rfd/0009-advanced-session-control.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+into these categories.
+
++ Keep a closer eye on contractor activity: Customers
++ Locking out the team during maintenance windows.
++ Terminating access for an employee before the TTL of Teleports cert.
```
Comment (webvictim) [inline rfd/0009-advanced-session-control.md:28]:
```suggestion
+ Terminating access for an employee before the TTL of Teleport's cert.
```

File: rfd/0009-advanced-session-control.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+foo@example.com   admin   Until 01/25/2019 14:22:31
+```
+
+## Implementation
+
+The `tctl users lock` and `tctl users unlock` commands will update `services.UserSpecV2`
+for the given user setting the Status fields to mark the user is locked or unlocked
+respectively.
+
+In `srv.ServerContext`, there exists periodic checks for disconnecting idle clients
```
Comment (webvictim) [inline rfd/0009-advanced-session-control.md:102]:
```suggestion
In `srv.ServerContext`, there are periodic checks for disconnecting idle clients
```

**Review by webvictim — APPROVED** [review]

**Comment by russjones** [comment]
I'm going to close this one because it's gone stale. https://github.com/gravitational/teleport/issues/3360 tracks details for user locking.

@andrejtokarcik please reference this RFD when you are writing the new RFD for user locking.

**Comment by andrejtokarcik** [comment]
Our RFD conventions aren't really clear in this regard, but wouldn't it be preferable to still merge this RFD as deprecated?

**Comment by webvictim** [comment]
If we want to keep a clear history of all RFD numbers then yes. I don't think we've done that so far though.

**Comment by awly** [comment]
`deprecated` state is intended for designs that were approved and/or implemented at one point, but are either reverted or superseded.
I sent https://github.com/gravitational/teleport/pull/6468 to clarify the policy.

**Pull request closed without merging**